File: gen_filtrations.py
#from weighted_filtration import get_weighted_pyramide_filtration
from flow_filtrations.general_flow_filtration import gen_flow_filtration
from flow_filtrations.lattice_neighbour_graph import create_lattice_neighoubr_graph
from data_reader import read_material_from_file
import joblib
from multiprocessing import Pool
import numpy as np

SAMPLE_NUM = 75
INPUT_NAME_TMPL = "../simulation_results/lmp_data_"
RADIUS = 2
HEIGHT = 3
OUTPUT_NAME_TMPL = "filtrations/flow_r" + str(RADIUS) + "_h" +str(HEIGHT) + "/rand/filtration_"
print(OUTPUT_NAME_TMPL)

def get_filtration(i):
    print(i)
    mat_data = read_material_from_file(INPUT_NAME_TMPL + str(i))
    grapher = lambda point, data: create_lattice_neighoubr_graph(point, data, HEIGHT, RADIUS)
    filtration =  gen_flow_filtration(mat_data, grapher)
    joblib.dump(filtration, OUTPUT_NAME_TMPL + str(i))

with Pool(4) as p:
    p.map(get_filtration, list(range(SAMPLE_NUM)))

    

File: gen_filtrations.jl
include("weighted.jl")
include("parser.jl")
using DelimitedFiles

function gen_out_name(i, z_multiplier, xy_multiplier, range)
    return ("filtrations/pyramide_" * string(z_multiplier) * "_" * string(xy_multiplier) * "_" * string(range) * "/rand/filtration_" * string(i))
end

function gen_filtration(i, z_multiplier, xy_multiplier, range)
    println("Case " * string(i))
    in_name = "../simulation_results/lmp_data_" * string(i)
    dt = read_material_from_file(in_name)
    sh = size(dt, 1)
    out = zeros(sh, sh, sh)

    for x in 1:sh
        for y in 1:sh
            for z in 1:sh
                out[x, y, z] = get_weighted_pyramide_filtration((x, y, z), dt)
            end
        end
    end

    writedlm(gen_out_name(i, z_multiplier, xy_multiplier, range), out)
end




for i in parse(Int64, ARGS[4]):74
    gen_filtration(i, parse(Float64, ARGS[1]), parse(Float64, ARGS[2]), parse(Int64, ARGS[3]))
end
